core/execution/market_snapshot.go

```diff
@@ -177,9 +177,12 @@ func NewMarketFromSnapshot(
 
 func (m *Market) getState() *types.ExecMarket {
 	rf := m.risk.GetRiskFactors()
-	var sp *num.Uint
+	var sp num.Numeric
 	if m.settlementDataInMarket != nil {
-		sp = m.settlementDataInMarket.Clone()
+		spp := m.settlementDataInMarket.Clone()
+		if spp != nil {
```

Comment:
How could `Clone` on `settlementDataInMarket` (which we've already confirmed to not be nil at this point) return nil?

Reply:
Yes, I forgot to remove that.
Corrected now.